pkg/generate/dockerfile/parser.go

```diff
@@ -0,0 +1,104 @@
+package dockerfile
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+	"unicode"
+)
+
+type dockerfile [][]string
+
+// Parser is a Dockerfile parser
+type Parser interface {
```

Comment:
Can't we just use the docker Dockerfile parser?
https://github.com/docker/docker/blob/master/builder/parser/parser.go